src/aiida/engine/processes/workchains/workchain.py

```diff
@@ -208,7 +208,7 @@ def _resolve_nested_context(self, key: str) -> tuple[AttributeDict, str]:
             #   (subclasses of AttributeDict) but after resolution of an Awaitable this will be the value itself
             # * assumption: a resolved value is never a plain AttributeDict, on the other hand if a resolved Awaitable
             #   would be an AttributeDict we can append things to it since the order of tasks is maintained.
-            if type(ctx) != AttributeDict:
```

Comment:
Same here

Reply:
`AttributeDict` is a class inheriting from `dict` so the `__eq__` function of the type is not overwritten so it is okay here